app/models/invite.rb

```diff
@@ -30,6 +30,7 @@ class ValidationFailed < StandardError; end
   validates_presence_of :invited_by_id
   validates :email, email: true, allow_blank: true
   validate :ensure_max_redemptions_allowed
+  validate :valid_domain
```

Comment:
We should only validate if the `domain` is going to be changed instead of validating every time the record is saved.